### PR TITLE
#64 Add checking count of commits

### LINKED ISSUE #64: Optimise app builds on CI
## Description
- repeating `chmod +x gradlew` in each job is redundant, try to find a way how to get rid of it,
- run locally `./gradlew build --dry-run` to check which tasks are executed during `build` command, 
  - do we need all of them?
  - should we keep all of them in one job?

### Make more readable regex
Written `regex` after time could be vague. So we should make it more readable.

### Optimize CI
At this moment our `build job` executed more than ten minutes it should be faster.

### Add checking count of commits 
We have the rule one `pull request` one `commit` therefore we should have a job that will check if a pull request has one commit.

### Helpful Links: 
- [DRY: REUSING CODE IN GITHUB ACTIONS](https://cardinalby.github.io/blog/post/github-actions/dry-reusing-code-in-github-actions/)

## Tasks
- [x] Make more readable regex
- [x] Optimize CI
- [x] Add checking count of commits 
- [x] Use Docker image
- [x] Fix checking commit message

### PR DESCRIPTION
**Pull request by Nataniel-Antosik**

### CODE REVIEW

File: .github/workflows/pull_request_verification.yml [hunk 1]
```diff
@@ -20,6 +20,14 @@ jobs:
       - name: Check correct commit message
         run: ./.github/scripts/commit/checkCommitMessage.sh
 
+  count-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Count Commits
+        run: ./.github/scripts/commit/checkCountCommits.sh "${{ github.event.pull_request.url }}" "${{ github.repository }}" "${{ secrets.GITHUB_TOKEN }}"
```
Comment (MarekMacko) [inline .github/workflows/pull_request_verification.yml]: Did you check if there is a direct access to number of commits via actions? Something like `github.event.pullrequest.commits` ?

Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml]: Yes, it's possible to use it therefore I did it and now it looks more easy then before :D

File: .github/workflows/pull_request_verification.yml [hunk 2]
```diff
@@ -20,6 +20,19 @@ jobs:
...
+      - name: PR commits
+        run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} ))" >> "${GITHUB_ENV}"
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: ${{ env.PR_FETCH_DEPTH }}
+      - name: Count Commits
+        run: ./.github/scripts/commit/checkCountCommits.sh "${{ env.PR_FETCH_DEPTH }}"
+
```
Comment (MarekMacko) [inline .github/workflows/pull_request_verification.yml:40]: Try to call script directly here, seems that flow of passing to script and retrieving the result, only to execute simple assertion looks like an overengineering

Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml:40]: Yes, I agree with you it looks like overengineering. I changed it 😄

File: .github/workflows/pull_request_verification.yml [hunk 3]
```diff
@@ -20,6 +20,23 @@ jobs:
       - name: Check correct commit message
         run: ./.github/scripts/commit/checkCommitMessage.sh
 
+  count-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: PR commits
+        run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} ))" >> "${GITHUB_ENV}"
```
Comment (MarekMacko) [inline .github/workflows/pull_request_verification.yml]: Is thins needed as a separate step? Are we able to merge it with `Count Commits` step?

Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml]: I will try it

Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml]: Done 😄

File: .github/workflows/pull_request_verification.yml [hunk 4]
```diff
@@ -20,6 +20,21 @@ jobs:
...
 
+  count-commits:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: ${{ github.event.pull_request.commits }}
+      - name: Count Commits
```
Comment (MarekMacko) [inline .github/workflows/pull_request_verification.yml]: `Assert commits`

File: .github/workflows/pull_request_verification.yml [hunk 5]
```diff
@@ -20,6 +20,21 @@ jobs:
...
+        run: |
+          if (( ${{ github.event.pull_request.commits }} > 1 )); then
```
Comment (MarekMacko) [inline .github/workflows/pull_request_verification.yml]: `github.event.pull_request.commits` could be extracted to a variable to not repeat this

File: .github/workflows/pull_request_verification.yml (hunk 4 above)
Comment (MarekMacko) [reply .github/workflows/pull_request_verification.yml]: or `Assert number of commits` - to indicate intention of that operation

Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml]: Done

File: .github/workflows/pull_request_verification.yml (hunk 5 above)
Comment (Nataniel-Antosik) [reply .github/workflows/pull_request_verification.yml]: Done

**Review by bartek977 — APPROVED** [review]

**Review by kmaslowiec — APPROVED** [review]

**Pull request merged**